examples/step-90/step-90.cc

```diff
@@ -0,0 +1,1271 @@
+          copy_data.cell_H1_error_sqr = cell_H1_error_sqr;
+
+          if (stabilization_scheme.needs_cell_worker())
+            {
+              typename DoFHandler<dim>::active_cell_iterator level_set_cell(
+                &(triangulation),
+                cell->level(),
+                cell->index(),
+                &level_set_dof_handler);
+              scratch_data.fe_values.reinit(cell);
+              scratch_data.level_set_fe_values.reinit(level_set_cell);
+              stabilization_scheme.evaluate_cell_worker(
+                locally_relevant_solution,
+                level_set,
+                cell,
+                scratch_data,
+                copy_data);
+            }
+        }
+    };
+
+    const auto copier = [&](const auto &copy_data) {
+      if (copy_data.cell_index < activeFE_indicator.size())
+        {
+          error_L2_sqr += copy_data.cell_L2_error_sqr;
+          error_H1_sqr += copy_data.cell_H1_error_sqr;
+          error_stab_sqr += copy_data.cell_stab_sqr;
+        }
+    };
+
+    ScratchData<dim> scratch_data(mapping,
+                                  fe_collection,
+                                  mesh_classifier,
+                                  level_set_dof_handler,
+                                  level_set,
+                                  surface_update_flags,
+                                  cell_quadrature,
+                                  quadrature_1D);
+
+    CopyData<dim> copy_data;
+
+    MeshWorker::mesh_loop(dof_handler.begin_active(),
+                          dof_handler.end(),
+                          cell_worker,
+                          copier,
+                          scratch_data,
+                          copy_data,
+                          MeshWorker::assemble_own_cells);
+    timer.stop();
+    pcout << "took (" << timer.wall_time() << "s)" << std::endl;
+  }
+
+  template <int dim>
+  void LaplaceBeltramiSolver<dim>::display_results()
+  {
+    const double error_L2 =
+      std::sqrt(Utilities::MPI::sum(error_L2_sqr, mpi_communicator));
+    const double error_semiH1 =
+      std::sqrt(Utilities::MPI::sum(error_H1_sqr, mpi_communicator));
+    const double error_stab =
+      std::sqrt(Utilities::MPI::sum(error_stab_sqr, mpi_communicator));
+
+    const int dof_handler_size           = dof_handler.n_dofs();
+    const int level_set_dof_handler_size = level_set_dof_handler.n_dofs();
+
+    const double iterations =
+      Utilities::MPI::max(number_of_iterations, mpi_communicator);
+
+    convergence_table.add_value("LevelSet dofs", level_set_dof_handler_size);
+    convergence_table.evaluate_convergence_rates(
+      "LevelSet dofs", ConvergenceTable::reduction_rate_log2);
+
+    convergence_table.add_value("Active dofs", dof_handler_size);
+    convergence_table.evaluate_convergence_rates(
+      "Active dofs", ConvergenceTable::reduction_rate_log2);
+
+    convergence_table.add_value("Iterations", iterations);
```

Comment:
I would put this at the end of the `solve()` function so that you don't need the member variable to store this between functions.